src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -240,6 +250,40 @@ private class BoolArrayEqualityComparer : IEqualityComparer<bool[]>
             }
         }
 
+        private class BlobTupleEqualityComparer : IEqualityComparer<Tuple<string, byte[], int>>
```

Comment:
You're right - I'll alter this. We were failing to compare Tuple objects as the same when the names in each were identical. I got a bit overly enthusiastic and compared all the contents.